service/controller/chart/v1/resource/tillermigration/create.go

```diff
@@ -0,0 +1,100 @@
+package tillermigration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+// EnsureCreated ensures tiller resources are deleted after all helm releases migrated into v3.
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Resource is used to remove tiller pod in tenant clusters.
```

Comment:
```suggestion
	// Resource is used to remove tiller. So for other charts we can skip this step.
```

I think this can be simpler and no need to mention tenant clusters. One day draughtsman will go away and then we can also clean control plane clusters.